mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp

```diff
@@ -424,3 +419,56 @@ Java_com_intel_oneapi_dal_table_HomogenTableImpl_cGetDoubleData(
     env->SetDoubleArrayRegion(newDoubleArray, 0, datasize, data);
     return newDoubleArray;
 }
+
+/*
+ * Class:     com_intel_oneapi_dal_table_HomogenTableImpl
+ * Method:    cEmptyTableInit
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_cEmptyTableInit
+  (JNIEnv *env, jobject) {
+      printf(" init empty HomogenTable \n");
+      homogen_table *h_table = new homogen_table();
+      homogenPtr tablePtr = std::make_shared<homogen_table>(*h_table);
+      saveShareHomogenPtrVector(tablePtr);
+      return (jlong)tablePtr.get();
+  }
+
+/*
+* Class:     com_intel_oneapi_dal_table_HomogenTableImpl
+* Method:    cAddHomogenTable
+* Signature: (Ljava/lang/Long;Ljava/lang/Long;)J
+*/
+JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_cAddHomogenTable
+ (JNIEnv *env, jobject, jlong mergedHomogenTablePtr,jlong homogenTablePtr){
+       printf("oneDal addHomogenTable \n");
+       homogen_table mergeTable = *((homogen_table *)mergedHomogenTablePtr);
+       homogen_table homogenTable = *((homogen_table *)homogenTablePtr);
+       if(mergeTable.has_data()){
+              printf("oneDal addHomogenTable mergeTable is not null \n");
+              const double* mergeData = mergeTable.get_data<double>();
+              const int mergeDatasize = mergeTable.get_column_count() * mergeTable.get_row_count();
+
+
+              const double* homogenData = homogenTable.get_data<double>();
+              const int homogenDatasize = homogenTable.get_column_count() * homogenTable.get_row_count();
+
+              long cRowCount = mergeTable.get_row_count() + homogenTable.get_row_count();
+              long cColCount = mergeTable.get_column_count();
+              double* row_doubles = new double[mergeDatasize + homogenDatasize];
+              for (std::int64_t i = 0; i < mergeDatasize; i++) {
+                  row_doubles[i] = mergeData[i];
+              }
+              for (std::int64_t i = 0; i < homogenDatasize; i++) {
+                  row_doubles[mergeDatasize + i] = homogenData[i];
+              }
+
+              homogen_table *resultTable = new homogen_table(
+                      row_doubles, cRowCount, cColCount, detail::empty_delete<const double>());
+              homogenPtr resultTablePtr = std::make_shared<homogen_table>(*resultTable);
```

Comment:
Same as above, should not use new.
```suggestion
              homogenPtr resultTablePtr = std::make_shared<homogen_table>(row_doubles, cRowCount, cColCount, detail::empty_delete<const double>());
```
```suggestion
              homogenPtr resultTablePtr = std::make_shared<homogen_table>(*resultTable);
```